main.go

```diff
@@ -100,44 +195,131 @@ func main() {
 			LAPIStream:      lapiStream,
 			UpdateFrequency: conf.CloudflareConfig.UpdateFrequency,
 			Wg:              &wg,
+			UpdatedState:    stateStream,
+			CFStateByAction: states,
+			Count:           Count,
+		}
+		if *onlySetup {
+			workerTomb.Go(func() error {
+				var err error = nil
+				defer func() {
+					workerTomb.Kill(err)
+					stateStream <- nil
+				}()
+
+				worker.CFStateByAction = nil
+				err = worker.Init()
+				if err != nil {
+					return err
+				}
+				err = worker.SetUpCloudflareIfNewState()
+				return err
+
+			})
+		} else if *delete {
+			workerTomb.Go(func() error {
```

Comment:
why not just running `deleteExistingIPList()` (not in a goroutine) and then `return` ?